src/project/internal/projectmigrator.cpp

```diff
@@ -66,12 +67,29 @@ Ret ProjectMigrator::migrateEngravingProjectIfNeed(engraving::EngravingProjectPt
     if (project->mscVersion() >= 400) {
         return true;
     }
-
     //! NOTE If the migration is not done, then the default style for the score is determined by the version.
     //! When migrating, the version becomes the current one, so remember the version of the default style before migrating
     project->masterScore()->style().setDefaultStyleVersion(ReadStyleHook::styleDefaultByMscVersion(project->mscVersion()));
     MigrationType migrationType = migrationTypeFromAppVersion(project->appVersion());
 
+    // there are a few things that need to be updated no matter which version the score is from (#10499)
+    // primarily, the differences made concerning barline thickness and distance
+    // these updates take place no matter whether or not the other migration options are checked
+    qreal sp = project->masterScore()->spatium();
+    Ms::MStyle* style = &project->masterScore()->style();
+    style->set(Ms::Sid::dynamicsFontSize, 10.0);
+    qreal doubleBarDistance = style->styleMM(Ms::Sid::doubleBarDistance);
+    doubleBarDistance -= style->styleMM(Ms::Sid::doubleBarWidth);
+    style->set(Ms::Sid::doubleBarDistance, doubleBarDistance / sp);
+    qreal endBarDistance = style->styleMM(Ms::Sid::endBarDistance);
+    endBarDistance -= (style->styleMM(Ms::Sid::barWidth) + style->styleMM(Ms::Sid::endBarWidth)) / 2;
+    style->set(Ms::Sid::endBarDistance, endBarDistance / sp);
+    qreal repeatBarlineDotSeparation = style->styleMM(Ms::Sid::repeatBarlineDotSeparation);
+    qreal dotWidth = project->masterScore()->scoreFont()->width(Ms::SymId::repeatDot, 1.0);
+    repeatBarlineDotSeparation -= (style->styleMM(Ms::Sid::endBarWidth) + dotWidth) / 2;
+    style->set(Ms::Sid::repeatBarlineDotSeparation, repeatBarlineDotSeparation / sp);
+    project->masterScore()->setResetDefaults();
```

Comment:
This whole new block of code doesn't look like it belongs to ProjectMigrator. Perhaps it should be moved to some public method of MasterScore

Reply:
I agree, though my main fear about that is that this only should happen once, on first open of an old score. I still have some work to do on this one so I'll keep searching for a better location for it